p2p/exchange.go

```diff
@@ -157,8 +167,15 @@ func (ex *Exchange[H]) Head(ctx context.Context, opts ...header.HeadOption[H]) (
 	)
 	for _, from := range peers {
 		go func(from peer.ID) {
+			_, newSpan := span.TracerProvider().Tracer("requesting peer").Start(
```

Comment:
This is actually an issue. It makes a new Tracer for every new request, while tracers are meant to be global. We easily use here the same client tracer for requesting peer spanss